sympy/functions/special/zeta_functions.py

```diff
@@ -214,6 +214,60 @@ def _eval_rewrite_as_zeta(self, z, s, a, **kwargs):
     def _eval_rewrite_as_polylog(self, z, s, a, **kwargs):
         return self._eval_rewrite_helper(polylog)
 
+    def _eval_is_finite(self):
+        z, s, a = self.args
+        if z.is_infinite or s.is_infinite or a.is_infinite:
+            # TODO: Write a valid implementation for infinite numbers.
+            return
+        if z == 0:
+            if a.is_zero and s.is_real:
+                return s.is_nonpositive
+            if a.is_nonzero:
+                return True
+            return
+        if a.is_integer and a.is_nonpositive:
+            if s.is_real:
+                return s.is_nonpositive
```

Comment:
Is this correct:
```python
In [21]: z, s, a = 10, -1, -10

In [22]: lerchphi(z, s, a).is_finite
Out[22]: True

In [23]: lerchphi(z, s, a).evalf()
...
ValueError: Lerch transcendent complex infinity
```

Reply:
Okay, I see what you mean. It does seem as if the function should be continuous there:
```
In [26]: lerchphi(10, -1, 10 + 1e-5).n()
Out[26]: -0.987655432098765

In [27]: lerchphi(10, -1, 10 - 1e-5).n()
Out[27]: -0.987653209876543
```

Reply:
Yes, that's another way. You checked for $a = 10$, which is not problematic, but the same can be done for the problematic point $a = -10$
```
In [26]: lerchphi(10, -1, -10 + 1e-5).n()
Out[26]: 1.23456679012452

In [27]: lerchphi(10, -1, -10 - 1e-5).n()
Out[27]: 1.23456901218742
```
which corresponds to the value given by WA of $\frac{100}{81} = 1.2345679012345679$

Reply:
Flint's `acb_dirichlet_lerch_phi` gives the same:
```python
In [22]: from flint import acb, ctx

In [23]: ctx.dps = 20

In [24]: acb(10).lerch_phi(-1, -10)
Out[24]: [1.23456790 +/- 9.62e-9] + [+/- 4.91e-9]j
```